libs/fail/fail.go

```diff
@@ -0,0 +1,78 @@
+package fail
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"strconv"
+)
+
+var callIndexToFail int
```

Comment:
note: we should probably rewrite this to avoid `callIndexToFail` being included in the runtime